src/python/pants/base/exception_sink.py

```diff
@@ -0,0 +1,78 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import logging
+import os
+import sys
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir, safe_open
+from pants.util.memo import memoized_classproperty
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  @memoized_classproperty
+  def instance(cls):
+    return cls()
+
+  def __init__(self, destination=None):
+    # TODO: what assumptions can we make about the current directory of a process? Does it have to
```

Comment:
The point of initializing it to something is just to increase the chances that even before it's been configured that we're still able to capture exceptions _somewhere_... so we're not really make assumptions, other than trying to increase our chances. The goal isn't every to actually use it without configuring it... rather to just still catch errors if we happen to fail that early.

Reply:
We can't, but that's not cromulent given the above.